[TEST] : 티켓 예매 등록 API 테스트

rhdtn311 opened the pull request:
## PR Description 🗒️
- 티켓 에몌 등록 API 테스트 코드를 작성하였습니다.

## 추가 사항 및 설명 ➕
- 예매 성공, 실패에 대한 서비스 레이어 단위 테스트, 통합 테스트, 컨트롤러 테스트 코드를 작성하였습니다.

## 수정 사항 및 이유 🔄

## To Reviewers 🙏

rhdtn311 committed:
[TEST] : 티켓 예매 등록 API 테스트

twotwobread committed:
[REFACTOR] : 변경된 Entity와 유저 로그인 Confilct 해소 및 merge (#125)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
close #71 
close #11 #27   
close #12 #13 #76 #88 #95 #107 
- 로그인/회원가입 소셜 로그인을 위한 기능 개발
- 유저 조회 관련 기능 개발 및 테스트 코드 작성.
- log 관련 설정
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- 소셜(구글) 로그인/회원가입 기능 개발 - [twotwobread](https://github.com/twotwobread),
[park-se-jun](https://github.com/park-se-jun) 페어프로그래밍
  - SpringSecufityConfig 설정
    - access deny handler 설정. 
- 정적 파일에 대해 인가 수행하지 않게 구성하고 User 권한을 가진 사람만 /api/v1/** url로 접근할 수 있게 구성.
- oauth2Login을 이용하게 구성하고 authorization endpoint,
authorizationRequestRepository, redirect endpoint, user info endpoint
설정.
    - success handler 설정.
  - 권한 확인 후 엑세스 거부되었을 때 실행될 OAuthAccessDeniedHandler  추가.
    - 상태코드 403과 함께 간단한 text 출력.
  - CookieOAuth2AuthorizationRequestRepository와 CookieUtil 추가.
    - request 정보로부터 Cookie를 가져오는 기능.
    - 쿠키를 저장하는 기능.
    - 쿠키를 삭제하는 기능.
  - CustomIOAuth2UserService와 CustomUserPrincipal 추가.
    - Resource 서버 (구글)로부터 User의 허용된 정보들을 가져오는 부분.
    - 해당 소셜 상의 id, 유저 이름, 유저 역할, 이메일 등의 정보를 가져옴.
    - 해당 정보로 OAuth2User의 구현체인 CustomUserPrincipal 내부 필드값으로 할당하여 반환.
  - success handler인 OAuth2AuthenticationSuccessHandler 추가
    -  실질적으로 우리의 서비스에 유저를 회원가입하는 부분.
- UserService의 join을 메소드를 이용해서 이미 유저가 존재하면 유저의 변경가능 정보인 user nickname을
업데이트하고 해당 유저 정보를 반환.
      - 유저가 존재하지 않으면 db에 저장하고 해당 유저 정보를 반환
- 반환된 유저 정보를 이용하여 access token과 refresh token 생성 및 기존 쿠키를 삭제하고 새로운
refresh token 정보를 쿠키에 저장.
    - 인증 성공 반환 정보로 access token을 반환.
  - TokenProvider 추가.
    - access, refresh 토큰을 생성하기 위한 로직 존재.
  - TokenAuthenticationFilter 추가.
- access toekn과 함께 요청이 들어올 경우 validate 수행하고 인증이 된것이 맞으면 Authentication을
생성하고 SpringSecutiryContext에 저장.
    - 인증되지 않은 경우 그냥 다음 filter 수행.
  - OAuthController
    - 만료된 토큰 refresh를 위해서 /refresh url 컨트롤러 추가. -> 갱신된 토큰 반환.
- 유저 상세 정보 조회 기능 - [twotwobread](https://github.com/twotwobread)
  - isDeleted 상태를 고려하여 repository, service, controller 구현.
  - 성공/실패 관련 test 코드 구현.
- 유저 페이징 조회 기능 - [twotwobread](https://github.com/twotwobread)
- 현재 페이지 번호가 가능한 마지막 페이지를 넘는 경우를 고려하여 repository, service, controller
구현.
  - 성공/실패 관련 test 코드 구현.
- log 관련 설정. - [twotwobread](https://github.com/twotwobread)
  - logback 로그 메시지 패턴 설정.
  - aws clock watch에 해당 로그가 쌓일 수 있게 설정. 

## To Reviewers 🙏
<!-- 
리뷰어들이 중점적으로 봐줬으면 하는 부분이나 
리뷰어들에게 추가적으로 하고 싶은 말을 적어주세요!! 
-->
- squash and merge 기능 때문에 커밋이 안보일꺼 같애서 최대한 자세히 적었습니다!!

---------

Co-authored-by: 박세준 <tpwns6693@gmail.com>
Co-authored-by: Kong <rhdtn311@naver.com>
Co-authored-by: rhdtn311 <68289543+rhdtn311@users.noreply.github.com>
Co-authored-by: So Hyeon <thgus345@naver.com>
Co-authored-by: 권성준 <tjdwns1865@naver.com>

twotwobread commented on src/test/java/com/prgrms/be/intermark/domain/ticket/controller/TicketControllerTest.java:
예매 등록 실패 관련해서는 문서화 해야하는 상황이 없을까요??

twotwobread commented on src/test/java/com/prgrms/be/intermark/domain/ticket/service/TicketServiceIntegrationTest.java:
assertThat도 static으로 등록해도 좋을 것 같습니다.

twotwobread reviewed (approved):
수고하셨습니다!!

park-se-jun committed:
[FIX] 71번을 develop으로 merge 하는 과정에서 발생한 error 해결 및 ci 추가 (#138)

<!-- 이슈와 PR 이름을 맞춰주세요!! -->
<!-- 여러 라인에 걸쳐 문장이 필요한 경우, 각 라인마다 "-" 를 이용해주세요 -->

## PR Description 🗒️
<!-- 
PR에 대한 간략한 설명을 적어주세요
ex)
- 로그인 기능 추가.
-->
- develop 으로 merge 하는 과정에서 발생한 error 해결
- 공통 application.yml 들 추가
- db와 환경변수를 고려하도록 ci 수정
- jacoco 관련 ci 추가
## 추가 사항 및 설명 ➕
<!-- 
[FEAT] / [ADD]
추가 사항과 이에 대한 설명을 적어주세요
ex)
- 예매 기능 추가
    - 
-->
- 코드 커버리지를 알려주는 댓글을 뭘 기준으로 보여주는 건지 잘 모르겠습니다. 아마 해당 pr에서 수정한 코드(테스트 코드
말고)들에 대해 보여주는것 같긴 함
- actions에서 jacoco report를 zip 로 다운 받을 수 있습니다.
- 이걸 가장 먼저 merge 할 수 있도록 해주세요.
## 수정 사항 및 이유 🔄
<!-- 
[CHORE] / [FIX] / [HOTFIX] / [REFACTOR]
수정 사항과 수정한 이유를 작성해주세요.
ex)
- 파라미터 타입을 Long에서 long으로 수정.
    - null이 포함되지 않고 단순 연산만 수행하여 수정.
-->

## To Reviewers 🙏
<!-- 
리뷰어들이 중점적으로 봐줬으면 하는 부분이나 
리뷰어들에게 추가적으로 하고 싶은 말을 적어주세요!! 
-->

park-se-jun reviewed (changes requested):
변경된 develop 과local에서 merge해 테스트가 여전히 잘 되는지 확인해 주시고, 잘 되면 카톡으로 다시 알려주세용

Seongjun-Kwon committed:
[FEAT] : 뮤지컬 등록 시 이미지 S3 에 저장하도록 구현

Seongjun-Kwon reviewed (approved):
고생하셨습니다~

thguss committed:
[TEST] : 스케줄 서비스 테스트 (#129)

park-se-jun committed:
[ADD] ec2 로의 배포 자동화와 관련된 workflow 작성 (#128)

rhdtn311 committed:
Merge branch 'feature/공태현#131' of https://github.com/prgrms-be-devcourse/BE-03-InterMark into test2

rhdtn311 committed:
[TEST] : 티켓 예매 테스트 코드 수정

- develop pull 받은 후 변경된 사항 수정
